### PR TITLE
43 us 021 implement unit tests

### PR DESCRIPTION
**Pull request by willayy**
Implemented 5 test suites testing the backend and some frontend pages.

Run by navigating to ```agile-app``` dir and running 
```bash 
npm test
```

### CODE REVIEW

File: agile-app/__tests__/testDataFetching.test.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+describe('Test datafetching', () => {
+  test('Text fetched crime data', async () => {
+    const crimeData = await getCrimeData();
+    expect(crimeData).not.toBeNull(); // Check that the data is not null
+    expect(crimeData.length).toBeGreaterThan(0); // Check that the data set is not empty
+    // Vlaidate that the data has the expected properties
+    expect(crimeData[0]).toHaveProperty('datetime'); 
+    expect(crimeData[0]).toHaveProperty('id');
+    expect(crimeData[0]).toHaveProperty('location');
```
Comment (Palpat3) [inline agile-app/__tests__/testDataFetching.test.ts:11]: Is there a reason for not testing the name and gps property inside the location?

File: agile-app/__tests__/testLineGraph.test.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+describe('LineGraph', () => {
+    it('renders without crashing', () => {
+        render(<LineGraph selectedOptionCrime="" selectedOptionLoc="" />);
+    });
+
+    it('renders the chart with correct attributes', () => {
+        const { getByTestId } = render(<LineGraph selectedOptionCrime="" selectedOptionLoc="" />);
+        const chart = screen.getByTestId('myChart');
+        expect(chart).toBeInTheDocument();
+        expect(chart).toHaveAttribute('width', '400');
```
Comment (Palpat3) [inline agile-app/__tests__/testLineGraph.test.tsx:14]: I think these are not up to date, the new size of the linegraph should be 600 by 600

**Review by Palpat3 — CHANGES_REQUESTED** [review]
Nothing that looks weird but some minor things I noticed and one test that is outdated

**Comment by jmarkusson** [comment]
Looks good, Why is there no tests on the heat map page?

File: agile-app/__tests__/testLineGraph.test.tsx (hunk 2 above)
Comment (willayy) [reply agile-app/__tests__/testLineGraph.test.tsx]: I fixed this

File: agile-app/__tests__/testDataFetching.test.ts (hunk 1 above)
Comment (willayy) [reply agile-app/__tests__/testDataFetching.test.ts:11]: Its already sufficiently validated

**Comment by willayy** [comment]
@Palpat3 I have fixed the points you raised

**Comment by willayy** [comment]
@jmarkusson Its because it uses functions that can't be used with jest

**Pull request merged**